xbmc/addons/gui/GUIDialogAddonInfo.cpp

```diff
@@ -329,29 +334,36 @@ void CGUIDialogAddonInfo::OnInstall()
   if (!g_passwordManager.CheckMenuLock(WINDOW_ADDON_BROWSER))
     return;
 
-  if (m_localAddon || !m_item->HasAddonInfo())
+  if (!m_item->HasAddonInfo())
     return;
 
-  std::string addonId = m_item->GetAddonInfo()->ID();
-  std::vector<std::pair<AddonVersion, std::string>> versions;
-
-  CAddonDatabase database;
-  if (!database.Open() || !database.GetAvailableVersions(addonId, versions) || versions.empty())
+  if (m_localAddon)
   {
-    CLog::Log(LOGERROR, "ADDON: no available versions of %s", addonId.c_str());
-    return;
+    std::string header = g_localizeStrings.Get(19098); // Warning!
+    std::string text =
+        StringUtils::Format(g_localizeStrings.Get(39028), m_localAddon->ID(),
+                            m_localAddon->Origin(), m_localAddon->Version().asString());
+    if (CGUIDialogYesNo::ShowAndGetInput(header, text))
+    {
+      m_silentUninstall = true;
+      OnUninstall();
+    }
+    else
+    {
+      return;
+    }
   }
 
-  int i = versions.size() == 1 ? 0 : AskForVersion(versions);
-  if (i != -1)
-  {
-    Close();
-    auto deps = CServiceBroker::GetAddonMgr().GetDepsRecursive(m_item->GetAddonInfo()->ID());
-    if (!deps.empty() && !ShowDependencyList(deps, false))
-      return;
+  std::string addonId = m_item->GetAddonInfo()->ID();
+  AddonVersion version = m_item->GetAddonInfo()->Version();
+  std::string origin = m_item->GetAddonInfo()->Origin();
```

Comment:
Think for here and some other places you can add a `const` before as them no more changed behind.

Reply:
changed them to const refs